span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanToLogRecordsTransformer.java

```diff
@@ -0,0 +1,92 @@
+package org.hypertrace.core.spannormalizer.jaeger;
+
+import static org.hypertrace.core.spannormalizer.constants.SpanNormalizerConstants.SPAN_NORMALIZER_JOB_CONFIG;
+
+import com.google.protobuf.util.Timestamps;
+import com.typesafe.config.Config;
+import io.jaegertracing.api_v2.JaegerSpanInternalModel;
+import io.jaegertracing.api_v2.JaegerSpanInternalModel.Span;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Transformer;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.hypertrace.core.datamodel.Attributes;
+import org.hypertrace.core.datamodel.LogEvent;
+import org.hypertrace.core.datamodel.LogEvents;
+import org.hypertrace.core.spannormalizer.util.AttributeValueCreator;
+
+public class JaegerSpanToLogRecordsTransformer
+    implements Transformer<byte[], Span, KeyValue<String, LogEvents>> {
+
+  private TenantIdHandler tenantIdHandler;
+  private SpanFilter spanFilter;
+
+  @Override
+  public void init(ProcessorContext context) {
+    Config jobConfig = (Config) context.appConfigs().get(SPAN_NORMALIZER_JOB_CONFIG);
+    tenantIdHandler = new TenantIdHandler(jobConfig);
+    spanFilter = new SpanFilter(jobConfig);
+  }
+
+  @Override
+  public KeyValue<String, LogEvents> transform(byte[] key, Span value) {
+    if (value.getLogsCount() == 0) {
+      return null;
+    }
+    Map<String, JaegerSpanInternalModel.KeyValue> tags =
+        value.getTagsList().stream()
```

Comment:
I think, this part is filtering based on missing tenant id or based on some configured criteria to drop spans, and it is common to both the transformers - `JaegerSpanToAvroRawSpanTransformer` and `JaegerSpanToLogRecordsTransformer`. So, I think, we should extract out as a common processing node.

```
KStream<byte[], Span> inputStream = (KStream<byte[], Span>) inputStreams.get(inputTopic);
KStream<byte[], Tuple<TraceId, Span>  filteredStream = inputStream.transform(xyz::new);
```
Both `JaegerSpanToAvroRawSpanTransformer`, `JaegerSpanToLogRecordsTransformer` can work on new `filteredStream`

Reply:
When you say `Tuple<TraceId, Span>` do you mean `Tuple<TenantId, Span>`? If it's not too much overhead, I'm fine with that. Else we can just rejigger the api of the tenantIdProvider to hide the common pre-processing.

Reply:
I have just encapsulated it in a class.